src/lib/local-storage-utils.js

```diff
@@ -0,0 +1,53 @@
+/* @flow */
+import constants from './constants';
+import generate from './generate-id';
+
+const { storage, sevenDays } = constants;
+
+/* Returns an existing cartId or null */
+export const getCartId = () => {
+    const storedValue = window.localStorage.getItem(storage.paypalCrCart);
+
+    if (storedValue) {
+        return JSON.parse(storedValue);
+    }
+
+    return null;
+};
+
+/* Sets a new cartId to expire in 7 days */
+export const setCartId = (cartId : string) => {
+    const storedValue = {
+        cartId,
+        createdAt: Date.now()
+    };
+
+    window.localStorage.setItem(storage.paypalCrCart, JSON.stringify(storedValue));
+
+    return storedValue;
+};
+
+/* Generates a random cartId that expires in 7 days */
+export const createNewCartId = () => {
+    const cartId = `${ generate.generateId() }`;
+
+    return setCartId(cartId);
+};
+
+/* Returns an existing cartId or creates a new one */
+export const getOrCreateCartId = () => {
+    return getCartId() || createNewCartId();
+};
+
+/* Generates a new cartId if the existing value is older than 7 days.
+Returns the current, up-to-date cartId in either case */
+export const resetExpiredCartId = () => {
```

Comment:
I think `getOrCreateValidCartId` is a better name for this, because you are not only reseting expired cart, but also returning a valid cartId.

Reply:
if what I said above makes sense (re. naming `getOrCreateValidCartId`), then the `getOrCreateCartId` function has little to no value.

Reply:
This is functionally different from 'getOrCreate' in that it checks the 'createdAt' value associated with the cartId.

The only reason 'getOrCreate' exists is to handle a potential programmer error. Ideally track should have nothing to do with initializing/setting up a cart id.